python/ray/data/_internal/execution/streaming_executor_state.py

```diff
@@ -104,6 +111,7 @@ def setup_state(op: PhysicalOperator) -> OpState:
         # Create state.
         op_state = OpState(op, inqueues)
         topology[op] = op_state
+        op.start(options)
```

Comment:
Shouldn't we defer this until the operator has started being dispatched to?

As is, this will currently cause all `MapOperators` using actors to create their actor pools before the operator has been dispatched any inputs, i.e. before it's ready to execute, which could lead to a lot of wasted resources and maybe even a liveness issue (although the latter shouldn't happen due to our base resource accounting). I was thinking that it would be better for an operator to defer creating its actor pool until it has been dispatched its first input or is at least eligible for dispatch.

Reply:
For streaming, the execution model is that all operators are instantiated and run concurrently, so I don't think this is a relevant concern. In any case, starting it all up from is the simpler/default approach we should start with.

Reply:
Sure, we can start with this! Starting up an actor pool for the last operator in a chain is taking away resources from the preceding operators, which might lead to some perf/liveness issues, but we can also solve that by pushing more of the actor pool autoscaling control into the executor.